### PR TITLE
feat: add type conflict checker to influx_inspect

### PR DESCRIPTION
**Pull request by davidby-influx**
adds two commands, `check-schema` and
`merge-schema`, to `influx_inspect`.
These test for field type conflicts
in all fields.idx beneath a directory
and merge the derived schemas if
"check-schema" has been run multiple
times on different directories

- `influx_inspect check-schema -path <data directory> -schema-file <schema output file> -conflicts-file <conflicts output file>`
- `influx_inspect merge-schema  schema-file <schema output file> -conflicts-file <conflicts output file>`

Defaults
- path: `.`
- schema-file: `schema.json`
- conflicts-file: `conflicts.json`

### CODE REVIEW

File: cmd/influx_inspect/type_conflicts/resolve_conflicts.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	flags.StringVar(&rc.ConflictsFile, "conflicts-file", "conflicts.json", "Filename conflicts data should be written to")
+
+	if err := flags.Parse(args); err != nil {
+		return err
+	}
+
+	return rc.nergeFiles(flags.Args())
+}
+
+func (rc *MergeFilesCommand) nergeFiles(filenames []string) error {
```
Comment (andrewcharlton) [inline cmd/influx_inspect/type_conflicts/resolve_conflicts.go:29]: Is "nerge" a typo?

File: cmd/influx_inspect/type_conflicts/schema.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+	s := NewSchema()
+	if err := s.Decode(f); err != nil {
+		return nil, fmt.Errorf("unable to decode schema file %q: %w", filename, err)
+	}
+	return s, nil
+}
+
+func (uf *UniqueField) String() string {
+	return fmt.Sprintf("%s.%s.%s.%s", uf.Database, uf.Retention, uf.Measurement, uf.Field)
```
Comment (andrewcharlton) [inline cmd/influx_inspect/type_conflicts/schema.go:59]: Is a "." character permissible in any of those?  i.e. could we have a two different UniqueField's with the same string?  Appreciate it's very unlikely, so can probably just be ignored.

**Review by andrewcharlton — APPROVED** [review]
Looks good to me (apart from the one typo!) :+1:

File: cmd/influx_inspect/type_conflicts/schema.go (hunk 2 above)
Comment (davidby-influx) [reply cmd/influx_inspect/type_conflicts/schema.go]: If the schema name is quoted, `.` is permitted. If unquoted, it is not.  I added quotes to the output with `%q` formatting, which is less human-readable but should handle all cases.

**Review by andrewcharlton — APPROVED** [review]

**Pull request merged**